src/seedu/addressbook/commands/Command.java

```diff
@@ -37,6 +37,16 @@
     protected List<? extends Assessment> relevantAssessments;
     protected List<? extends ReadOnlyExam> relevantExams;
     private int targetIndex = -1;
+    private int targetExamIndex = -1;
+
+    /**
+     * Signals that the target exam index is out of bounds of the last viewed exams listing
+     */
+    public static class ExamIndexOutOfBoundsException extends IndexOutOfBoundsException {
+        public ExamIndexOutOfBoundsException(String message) {
+            super(message);
+        }
+    }
```

Comment:
Is there a reason to create a new exception rather than using the default IndexOutOfBoundsException ?

Reply:
yes to make it clear which index is the problem